src/new-components/modal/modal.tsx

```diff
@@ -355,12 +355,14 @@ export function ModalFooter({
  * @see ModalFooter
  */
 export function ModalActions({ children, ...props }: ModalFooterProps) {
-    return (
+    const childItems = React.Children.toArray(children).filter((child) => Boolean(child))
+
+    // If all child items are null/undefined, then don't show the footer at all and alleviate any
+    // excess spacing within the modal.
+    return childItems.length === 0 ? null : (
         <ModalFooter {...props}>
             <Inline align="right" space="large">
-                {React.Children.map(children, (child) => (
-                    <div>{child}</div>
-                ))}
+                {childItems.map((child, i) => (child ? <div key={i}>{child}</div> : null))}
```

Comment:
I don't think we need to check here for falsy values as we filter them out above already. All `child` should be truthy 👍

---

A question for @gnapse: why do we need to wrap the children in `div`s in the first place? Shouldn't `Inline` work with anything we throw at it? And if not should we pass this burden to the user?

Reply:
> I don't think we need to check here for falsy values as we filter them out above already. All child should be truthy 👍

Say hello to the original fix 😄 I'll remove that check now.

Reply:
> why do we need to wrap the children in `div`s in the first place? Shouldn't `Inline` work with anything we throw at it? And if not should we pass this burden to the user?

Nope. This was a compromise to avoid having `Inline` and `Stack` generate excessive markup all the time. See https://github.com/Doist/reactist/wiki/Simplified-markup-of-the-Stack-and-Inline-components

Reply:
Makes sense and awesome that we have documentation on this ❤️